result/overall.go

```diff
@@ -41,6 +41,13 @@ type PartialResult struct {
 	defaultStateSet    bool // nolint: unused
 }
 
+func NewPartialResult() PartialResult {
```

Comment:
Look alright. But could you add a doc string for the function, maybe describe why the defaults are what they are?